### PR TITLE
Update file timestamps to avoid redundant re-syncs (Fixes #22)

### LINKED ISSUE #22: Copied files do not get their timestamps updated
Hi!

This package is amazing, with all its dependencies.
There's one thing that's missing, and probably a bug - the synced files do not get their timestamps modified after being written.
The outcome will be that if you run the same sync operation again - they will be re-written.

You need to call `Fs.utimes` on it to update the timestamps.

### PR DESCRIPTION
**Pull request by danielgindi**

### CODE REVIEW

**Comment by stefanpenner** [comment]
Could you provide a failing test?

**Comment by danielgindi** [comment]
Note that for folders the `mtime` will change as files are added to the folders.
So there's not much point there, unless you will add an mtime update after the files are created/deleted there.
But that will be both overly complicated and kind of idiotic, because the folder's mtime will not really reflect the truth. If there's a sync, and some files have changed - ignoring some other files, the directories are different and have the same mtime - not a good idea.

**Comment by danielgindi** [comment]
I don't really get why that test fails on *some* machines. Is `utimes` not working on some of those envs?

**Comment by danielgindi** [comment]
Okay so this is a failure in very old nodejs versions:
https://github.com/nodejs/node/issues/2069

You are targetting versions below node 6, which are not even supported anymore.
If you'll upgrade the testing envs, you'll also get the benefit of being able to upgrade the codebase to more modern syntax :-)

**Comment by stefanpenner** [comment]
Ya, I’m down with a major version bump. So feel free to bump the engine support to >= 8 (as node 6 is on its way out)

**Comment by stefanpenner** [comment]
(I’m also fine with es syntax/feature upgrade if you are interested)

**Comment by danielgindi** [comment]
@stefanpenner This should also be a major version bump

File: .travis.yml [hunk 1]
```diff
@@ -2,10 +2,8 @@ sudo: false
 
 language: node_js
 node_js:
-  - "0.10"
-  - "0.12"
-  - "4"
-  - "5"
+  - "8.15"
```
[the remark this answers is not in the document]
Comment (danielgindi) [reply .travis.yml]: Done

File: .travis.yml [hunk 2]
```diff
@@ -2,10 +2,8 @@ sudo: false
 
 language: node_js
 node_js:
```
[the remark this answers is not in the document]
Comment (danielgindi) [reply .travis.yml]: Done

**Comment by danielgindi** [comment]
@stefanpenner Done. It won't automatically resolve the "requested changes" as one of them is on a different file.

**Comment by stefanpenner** [comment]
Tests appear to be failing on OSX

```
 1) TreeSync fixtures/one nothing -> populated validate mtime stays the same:

      AssertionError: expected 1532724030000 to equal 1532724030412
      + expected - actual

      -1532724030000
      +1532724030412

      at Context.<anonymous> (tests/index.js:63:34)
```

**Comment by stefanpenner** [comment]
NM is was an old 8.x that was failing, latest 8.15 works fine.

**Comment by danielgindi** [comment]
I have just tested it now:
```

  TreeSync
    fixtures/one
      nothing -> populated
        ✓ file content
        ✓ validate mtime stays the same
      existing empty directory -> populated
        ✓ file content
      rmdir operation is sync
        ✓ immediately reflects deletions
      input(same) -> input(same)
        ✓ has stable output (mtime, size, mode, relativePath)
      .sync
        ✓ returns a list of changed files
      input(same) -> input(same + newFile)
        ✓ has stable output (mtime, size, mode, relativePath)
      output populated -> input with a changed file
        ✓ should update changed files on initial build
      input(same) -> input(same - file)
        ✓ has stable output (mtime, size, mode, relativePath)
      validate walk-sync options
        ✓ should ignore files/folders it is told to ignore
        ✓ should only include globs it is told to include

  eslint
    ✓ should have no errors in tests/index.js (359ms)
    ✓ should have no errors in tests/linting.js
    ✓ should have no errors in index.js (74ms)


  14 passing (649ms)
```

on macOS 10.14.12.

What is your env exactly?

**Comment by danielgindi** [comment]
> NM is was an old 8.x that was failing, latest 8.15 works fine.

Oh. That's why I wanted to specifically go with the LTS version, not just 8.x. Just remembered I was looking at the specific version where LTS started...
But the chance of someone using a non LTS version of node 8.x is very small.

**Comment by stefanpenner** [comment]
released as v2.0.0 🎉

**Pull request merged**